### DISCUSSION TITLE
Using OpenCV Grabber  with YARP

### DISCUSSION THREAD

**Discussion by apaikan** [robotology/community#211] [Show and tell]
## Using OpenCV Grabber  with YARP
This is a quick tutorial of how to use Yarp `opencv_grabber` to 
1. Capture image for your **webcam** and play on a YARP port 
2. Play a **video file** on a YARP port 
3. Play a **images sequence** on a YARP port 

### Enable `opencv_grabber` device in YARP  
```
$ cd yarp/build
$ cmake -DCREATE_DEVICE_LIBRARY_MODULES=ON  -D=ENABLE_yarpmod_opencv_grabber=ON ../
$ make 
```

### The `opencv_grabber` device options 
- `--name <value>`        set the output port name (default is `/grabber`) 
- `--camera <value>`     set the number of the capture device (e.g. 0, for the first camera) 
- `--movie  <value>`       set the name of the video/image sequence files instead of camera for capturing
- `--framerate <value>`  set the desired frame rate  
- `--width <value>`         set the width of the image
- `--height <value>`       set the height of the image
- `--loop`                        enable image looping when capturing from  video/image sequence files
- `--flip_x`,                      flip the image along the x-axis
- `--flip_y`,                        if present, flip the image along the y-axis
- `--transpose`                  rotate the image along of 90 degrees

### Example 1:  Capturing from camera (default camera frame size and rate)
```
$ yarpdev --device opencv_grabber --name /image:o --camera 0 
```

### Example 2:  Capturing from a video file (avi, mp4, ..) 
```
$ yarpdev --device opencv_grabber --movie myfile.avi --framerate 30 --loop
```

### Example 3:  Capturing from images sequence (e.g., `yarpdatadumper`)   
```
$ yarpdev --device opencv_grabber  --framerate 30 --movie %8d.ppm --loop
```

**Comment by towardthesea** [comment]
Should we mention this tutorial as a temporary solution for *python* with *yarpdataplayer* :smiley:
